File: golang/primosRecursivos.go
package main

import "math"

func primosRecursivos(numero int) []int  {
	if numero < 2 {
		return nil
	}

	primos := primosRecursivos(numero - 1)
	ehPrimo := true
	limite := int(math.Sqrt(float64(numero)))

	for i := 2; i <= limite; i++ {
		if numero % i == 0 {
			ehPrimo = false
			break
		}
	}
	if ehPrimo {
		primos = append(primos, numero)
	}

	return primos
}

File: python/primosRecursivos.py
import math


def primosRecursivos(numero: int) -> list[int]:
    if numero < 2:
        return []

    primos = primosRecursivos(numero - 1)
    ehPrimo = True
    limite = int(math.sqrt(numero))

    for i in range(2, limite + 1):
        if numero % i == 0:
            ehPrimo = False
            break

    if ehPrimo:
        primos.append(numero)

    return primos


if __name__ == "__main__":
    print(primosRecursivos(10))
